### ISSUE TITLE
evaluate() returning weird result

### ISSUE THREAD

**Issue by robertcrowe** [tflearn/tflearn#378]
I'm trying to create a regression model to predict a single continuous value.  I have 43 features.  I can fit the model, but when I use evaluate() it returns simply [1.0], which makes no sense to me.  Here's my code:

```
net = tflearn.input_data(shape=[None, 43])
net = tflearn.fully_connected(net, 86, activation='relu')
net = tflearn.fully_connected(net, 43, activation='relu')
net = tflearn.fully_connected(net, 1, activation='linear')
net = tflearn.regression(net, optimizer='sgd', loss='mean_square')

model = tflearn.DNN(net)
model.fit(train_features, train_labels, n_epoch=1000, show_metric=True, snapshot_epoch=False)

result = model.evaluate(test_features, test_labels)
```
I tried adding metric='R2' to regression() but it complained that the input and output vectors need to be the same shape, which also seems very weird.  What am I doing wrong?  Also, could we get some more regression examples?  The current examples are almost all classification, but there are a lot of use cases for predicting continuous values.

**Comment by robertcrowe** [comment]
No response in a week?

**Comment by aymericdamien** [comment]
Hi, the reason it always output [1.0] is due to the fact that accuracy is not suitable for evaluating continuous values, it was designed for classification. R2 can be used for continuous values, but it requires X and Y to have same shape. 
For regression over continuous values, loss itself is already a meaningfull metric, but according to your specific case, you can write you own metric (see https://github.com/tflearn/tflearn/blob/master/tflearn/metrics.py#L64 for an example, and use that class into regression layer `metric` argument).

**Comment by robertcrowe** [comment]
Thanks for the response, but I'm still mystified by this part:
> R2 can be used for continuous values, but it requires X and Y to have same shape. 

I'm pretty sure that the Coefficient of Determination (R2) only requires the labels and predicted values (y and y_hat) to have the same shape, not the labels and features (x and y).

In sklearn it would be perfectly valid to do this:

```
>>> from sklearn.metrics import r2_score
>>> y_true = [3, -0.5, 2, 7]
>>> y_pred = [2.5, 0.0, 2, 8]
>>> r2_score(y_true, y_pred)  
0.948...
>>> y_true = [[0.5, 1], [-1, 1], [7, -6]]
>>> y_pred = [[0, 2], [-1, 2], [8, -5]]
>>> r2_score(y_true, y_pred, multioutput='variance_weighted')  
0.938...

```
For example, see:
http://scikit-learn.org/stable/modules/generated/sklearn.metrics.r2_score.html

**Comment by aymericdamien** [comment]
Yes indeed! I can't recall why it was implemented that way... I adapted the r2 metric. You can reinstall tflearn, and it should works. Thanks for pointing out that error!

**Comment by robertcrowe** [comment]
I'm not sure if I reinstalled it correctly, but I'm still getting an exception:

> Exception, R2 metric requires Inputs and Targets to have same shape.

To reinstall I did:

`% sudo pip install --upgrade tflearn`

Did I do it wrong?

**Comment by robertcrowe** [comment]
I'm getting R2 results of greater than 1.0, but I don't think that should be possible.  I would tend to think that it's overfitting, but if the number doesn't make sense it's hard to know.  Do you see any way that it should ever be > 1?

**Comment by robertcrowe** [comment]
It looks like there is the same problem with weighted R2:
> Exception: Weighted R2 metric requires Inputs and Targets to have same shape.

And I'm also still getting R2 values > 1, which don't make sense.  I'm beginning to wonder about the R2 code.

**Comment by deepakanandece** [comment]
I have a trained model for classification to four classes. I have used categorical crossentropy as metric. Still the prediction are stuck to one label for all example. If anyone knows the reason please respond.

**Comment by DollarAkshay** [comment]
Having a very similar problem here :/
Getting the same output for everything.

**Comment by Iskander2015** [comment]
Having R2>1, how to fix it?
------------------------------
tf.reset_default_graph()
# Build neural network
net = tflearn.input_data(shape=[None, len(train_x[0])])
net = tflearn.fully_connected(net, 8)
net = tflearn.fully_connected(net, 8)
net = tflearn.fully_connected(net, len(train_y[0]), activation='linear')
net = tflearn.regression(net, optimizer='adam', loss='mean_square', metric='R2', learning_rate=0.3)

m = tflearn.DNN(net)
m.fit(train_x, train_y, n_epoch=200, show_metric=True, snapshot_epoch=False)

Training Step: 48882  | total loss: 129897152.00000 | time: 0.226s
| Adam | epoch: 200 | loss: 129897152.00000 - R2: 8.7829 -- iter: 08128/15673
 | time: 0.433s
| Adam | epoch: 200 | loss: 47616004.00000 - R2: 3.2840 -- iter: 15673/15673

**Comment by ChemicalXandco** [comment]
@Iskander2015 Your loss is through the roof. `0.001` is a good lr for adam, don't know why you have 3, Thats quite extreme.
